plugins/modules/yarn.py

```diff
@@ -206,16 +208,23 @@ def _exec(self, args, run_in_check_mode=False, check_rc=True):
                 cmd.append(self.registry)
 
             # If path is specified, cd into that path and run the command.
+
+            path = self.path
             cwd = None
-            if self.path and not self.globally:
-                if not os.path.exists(self.path):
+
+            if self.globally and unsupported_with_global:
+                _rc, out, _err = self.module.run_command(self.executable + ['global', 'dir'], check_rc=check_rc)
+                path = out.strip()
```

Comment:
This does mean a `global=true` param will override the `path` param, but that was previously true anyway (since `yarn global` from any working directory would operate on the global packages).